lib/system/mmdisp.nim

```diff
@@ -46,7 +46,7 @@ const
   PageSize = 1 shl PageShift
   PageMask = PageSize-1
 
-  MemAlign = 8 # also minimal allocatable memory block
+  MemAlign = 16 # also minimal allocatable memory block
```

Comment:
This adds quite some overhead to the Nim compiler itself. The compiler doesn't care about SSE data anyway. Not every piece of software is a game.

Reply:
I explained earlier why I changed the value. If you really really don't want it, I can revert it back. But it basically means that SSE can't be used at all. Not every software that needs SSE is a game.

Reply:
I suggest the following approach. 
This PR gives automatic flow of alignment type info. Now we can do dynamic alignment in allocator instead of static 8 or 16 bytes. This is much better than C/C++ approach that always overallocates to 16 bytes in 64bit mode because malloc has no access to type info.

For example if struct consists only of chars then it will get alignment of 1 and allocator can take advantage of this info and provide most efficient in terms memory usage allocation. 
If object contains SSE it will get 16 byte alignment, if AVX it will get 32 byte alignment. 
`MemAlign` no longer needs to be a constant.

I suggest passing the alignment info to allocator. In first implementation it can still overallocate upwards. I suggest allocator to have API similar to this one https://github.com/microsoft/mimalloc/blob/acb03c54971c4b0a43a6d17ea55a9d5feb88972f/src/alloc-aligned.c#L17

```nim
proc rawAllocAlignedAt(a: var MemRegion, size: int, offset = 0, align = MemAlign): pointer =
```
In this way allocator can allocate aligned, offset aligned (required because of FreeCell header).